plugin/ochttp/server_test.go

```diff
@@ -273,3 +276,88 @@ func TestHandlerProxiesHijack_HTTP2(t *testing.T) {
 		t.Errorf("got = %q; want = %q", g, w)
 	}
 }
+
+func TestEnsureTrackingResponseWriterSetsStatusCode(t *testing.T) {
+	// Ensure that the trackingResponseWriter always sets the spanStatus on ending the span.
+	// Because we can only examine the Status after exporting, this test roundtrips a
+	// couple of requests and then later examines the exported spans.
+	// See Issue #700.
+	spanInspector := new(spanExporter)
+	trace.RegisterExporter(spanInspector)
+	defer trace.UnregisterExporter(spanInspector)
+
+	tests := []struct {
+		res  *http.Response
+		want trace.Status
+	}{
+		{res: &http.Response{StatusCode: 200}, want: trace.Status{0, `"OK"`}},
+		{res: &http.Response{StatusCode: 500}, want: trace.Status{2, `"UNKNOWN"`}},
+		{res: &http.Response{StatusCode: 403}, want: trace.Status{7, `"PERMISSION_DENIED"`}},
+		{res: &http.Response{StatusCode: 401}, want: trace.Status{16, `"UNAUTHENTICATED"`}},
+		{res: &http.Response{StatusCode: 429}, want: trace.Status{8, `"RESOURCE_EXHAUSTED"`}},
+	}
+
+	for i, tt := range tests {
+		span := trace.NewSpan("testing", nil, trace.StartOptions{Sampler: trace.AlwaysSample()})
+		ctx := trace.WithSpan(context.Background(), span)
+		prc, pwc := io.Pipe()
+		go func() {
+			pwc.Write([]byte("Foo"))
+			pwc.Close()
+		}()
+		inRes := tt.res
+		inRes.Body = prc
+		tr := &traceTransport{base: &responseBase{res: inRes}}
+		req, err := http.NewRequest("POST", "https://example.org", bytes.NewReader([]byte("testing")))
+		if err != nil {
+			t.Errorf("#%d: NewRequest error: %v", i, err)
+			continue
+		}
+		req = req.WithContext(ctx)
+		res, err := tr.RoundTrip(req)
+		if err != nil {
+			t.Errorf("#%d: RoundTrip error: %v", i, err)
+			continue
+		}
+		_, _ = ioutil.ReadAll(res.Body)
+		res.Body.Close()
+	}
+
+	// And then finally, we'll examine the exported spanData
+	// Firstly ensure that we have exactly the expected number.
+	if g, w := len(spanInspector.store), len(tests); g != w {
+		t.Errorf("Exported SpanData length: got %d want %d", g, w)
+	}
+
+	// And then, let's assert them, in order.
+	for i, tt := range tests {
+		got, want := spanInspector.store[i].Status, tt.want
+		if got != want {
+			t.Errorf("SpanData: #%d:\ngot =  (%#v)\nwant = (%#v)", i, got, want)
+		}
+	}
+}
+
+type spanExporter struct {
+	sync.Mutex
+	t     *testing.T
+	store []*trace.SpanData
+}
+
+var _ trace.Exporter = (*spanExporter)(nil)
+
+func (se *spanExporter) ExportSpan(sd *trace.SpanData) {
+	se.Lock()
+	se.store = append(se.store, sd)
+	se.Unlock()
+}
+
+type responseBase struct {
```

Comment:
I'd name these with a test prefix.

type testTransport struct

Reply:
Roger that, thanks.